pkg/logs/config/config.go

```diff
@@ -23,11 +23,11 @@ func DefaultSources() []*LogSource {
 	tcpForwardPort := LogsAgent.GetInt("logs_config.tcp_forward_port")
 	if tcpForwardPort > 0 {
 		// append source to collect all logs forwarded by TCP on a given port.
-		tcpForwardSource := NewLogSource("tcp_forward", &LogsConfig{
+		source := NewLogSource("tcp_forward", &LogsConfig{
 			Type: TCPType,
 			Port: tcpForwardPort,
 		})
-		sources = append(sources, tcpForwardSource)
+		sources = append(sources, source)
```

Comment:
Question: Why don't we have the same deadlock problem with this source?

Reply:
This a good question, because we call `agent start` then `add default sources` which ensures that all consumers are already registered to source channels.
https://github.com/DataDog/datadog-agent/blob/master/pkg/logs/logs.go#L46-L49
https://github.com/DataDog/datadog-agent/blob/master/pkg/logs/logs.go#L52-L54
I'd like to avoid creating sources manually from outside the logs-scheduler, I'll create a ticket for this.

For the deadlock
What happened is:
```
c := make(chan struct{})
c <- struct{}
go func() {
  <- c
}()
```
And we fixed it with:
```
c := make(chan struct{})
go func() {
  <- c
}()
c <- struct{}
```

Reply:
ok thanks